draft-ietf-privacypass-architecture.md

```diff
@@ -544,6 +545,76 @@ and guidelines on how to deploy the protocol to minimize any privacy impacts.
 Any extension to the Privacy Pass protocol MUST adhere to the guidelines
 specified in {{issuer-role}} for managing Issuer public key data.
 
+## Information Flow {#flow}
+
+The end-to-end run of redemption and issuance protocol involves information
+flowing between Issuer, Origin, Client, and Attester. That information can
+have implications on the privacy goals that Privacy Pass aims to provide
+as outlined in {{privacy-and-trust}}. In this section, we describe the flow
+of information between each party. How this information affects the privacy
+goals in particular deployment models is further discussed in {{deployment}}.
+
+### Token Challenge Flow {#challenge-flow}
+
+To use Privacy Pass, Origins choose an Issuer from which they are willing to
+accept tokens. Origins then construct a token challenge using this specified
+Issuer and the redemption context it shares with the Client. This token
+challenge is then delivered to a Client. The token challenge conveys
+information about the Issuer and the redemption context, such as whether the
+Origin desires a per-Origin or cross-Origin token.
+
+### Attestation and Issuance Flow {#issue-flow}
+
+Clients use the issuance protocol to produce a token bound to a token
+challenge. In doing so, there are several ways in which the issuance protocol
+contributes information to the attestation context and issuance context.
+For example, a token request may contribute information to the attestation
+or issuance contexts as described below.
+
+- Issuance protocol. The type of issuance protocol can contribute information
+about the Issuer's capabilities to the attestation or issuance contexts.
+- Issuer identity or configuration. Information about the Issuer, such as its
+identity or the public key used to validate tokens it creates, can be revealed
+during issuance and contribute to the attestation or issuance contexts.
+- Attestation information. The issuance protocol can contribute information to
+the issuance context based on what attestation procedure the Issuer uses to
+trust a token request. In particular, a token request that is validated by
+a given Attester means that the Client which generated the token request must
+be capable of the completing the designated attestation procedure.
```

Comment:
This is necessary information, isn't it?  We really do need the Issuer to be convinced that the Client passed whatever tests the Attester set.

The other aspect to consider is what unwanted information might be revealed here.  If the Attester would accept a Client that meets criterion A, B, or C, then it should not be the case that the Issuer learns which of A, B, or C was used.  That seems fine.  However, if there are multiple valid Attesters, I don't see a way to avoid the Issuer learning which Attester was used.  The choice of Attester therefore says something about the client.  There is a transitory flow of information here though: if Attester A supports method A and Attester B supports method B, then the choice of Attester indirectly reveals the choice of method.

Reply:
This list isn't meant to be exhaustive, as I don't expect us to be able to list all exhaustive things. It contains the what seem to me to be the obvious things that might be revealed, and the type of attestation that's done is one of those obvious things. It is necessarily revealed to the issuer through the process here.

Reply:
Obvious or not, it is still worth spelling out explicitly.  These are points where information leaks as a direct result of the architectural choices.